Mathlib/MeasureTheory/Measure/Portmanteau.lean

```diff
@@ -564,23 +568,80 @@ theorem tendsto_of_forall_isOpen_le_liminf_nat {μ : ProbabilityMeasure Ω}
   convert obs
   simp only [Function.comp_apply, ProbabilityMeasure.ennreal_coeFn_eq_coeFn_toMeasure]
 
-/-- One implication of the portmanteau theorem:
-If for all open sets G we have the liminf condition `μ(G) ≤ liminf μsₙ(G)`, then the measures
-μsₙ converge weakly to the measure μ. Formulated here for countably generated filters. -/
+/-- One implication of the portmanteau theorem: if for all open sets `G` we have the liminf
+condition `μ(G) ≤ liminf μsₙ(G)`, then the measures `μsₙ` converge weakly to the measure `μ`.
+
+This lemma uses a coercion from `ProbabilityMeasure` to `Measure` in the hypothesis.
+See `tendsto_of_forall_isOpen_le_liminf` for the version without coercion. -/
+theorem tendsto_of_forall_isOpen_le_liminf' {ι : Type*} {μ : ProbabilityMeasure Ω}
+    {μs : ι → ProbabilityMeasure Ω} {L : Filter ι} [L.IsCountablyGenerated]
+    (h_opens : ∀ G, IsOpen G → (μ : Measure Ω) G ≤ L.liminf (fun i ↦ (μs i : Measure Ω) G)) :
+    L.Tendsto (fun i ↦ μs i) (𝓝 μ) := by
+  apply Filter.tendsto_of_seq_tendsto fun u hu ↦ ?_
+  apply tendsto_of_forall_isOpen_le_liminf_nat' fun G hG ↦ ?_
+  exact (h_opens G hG).trans (liminf_le_liminf_of_le hu)
+
+/-- One implication of the portmanteau theorem: if for all open sets `G` we have the liminf
+condition `μ(G) ≤ liminf μsₙ(G)`, then the measures `μsₙ` converge weakly to the measure `μ`.
+Formulated here for countably generated filters. -/
 theorem tendsto_of_forall_isOpen_le_liminf {ι : Type*} {μ : ProbabilityMeasure Ω}
     {μs : ι → ProbabilityMeasure Ω} {L : Filter ι} [L.IsCountablyGenerated]
     (h_opens : ∀ G, IsOpen G → μ G ≤ L.liminf (fun i ↦ μs i G)) :
     L.Tendsto (fun i ↦ μs i) (𝓝 μ) := by
-  apply Filter.tendsto_of_seq_tendsto (fun u hu ↦ ?_)
-  apply tendsto_of_forall_isOpen_le_liminf_nat (fun G hG ↦ ?_)
-  apply (h_opens G hG).trans
+  apply Filter.tendsto_of_seq_tendsto fun u hu ↦ ?_
+  apply tendsto_of_forall_isOpen_le_liminf_nat fun G hG ↦ (h_opens G hG).trans ?_
   change _ ≤ atTop.liminf ((fun i ↦ μs i G) ∘ u)
   rw [liminf_comp]
   refine liminf_le_liminf_of_le hu (by isBoundedDefault) ?_
   exact isBoundedUnder_of ⟨1, by simp⟩ |>.isCoboundedUnder_ge
 
 end le_liminf_open_implies_convergence
 
+section Closed
+
+variable {Ω ι : Type*} {mΩ : MeasurableSpace Ω} [TopologicalSpace Ω] [OpensMeasurableSpace Ω]
+    {μ : ProbabilityMeasure Ω} {μs : ι → ProbabilityMeasure Ω}
+    {L : Filter ι} [L.IsCountablyGenerated]
+
+/-- One implication of the portmanteau theorem: if for all closed sets `F` we have the limsup
+condition `limsup μsₙ(F) ≤ μ(F)`, then the measures `μsₙ` converge weakly to the measure `μ`.
+Formulated here for countably generated filters.
+
+This lemma uses a coercion from `ProbabilityMeasure` to `Measure` in the hypothesis.
+See `tendsto_of_limsup_measure_closed_le` for the version without coercion. -/
+lemma tendsto_of_limsup_measure_closed_le'
```

Comment:
I think I prefer the naming scheme in the open situation, which would give
```suggestion
lemma tendsto_of_forall_isClosed_limsup_le'
```
which matches the statement more closely (notably, `isClosed` instead of `closed`) while not being less readable. Same thing below.